packages/core/src/components/Dropdown/Dropdown.tsx

```diff
@@ -1,77 +1,615 @@
+      containerRef != null ? [...containerRef.getElementsByTagName("li")] : [];
+    listItems.every((listItem) => {
+      if (listItem.tabIndex >= 0) {
+        listItem.focus();
+        return false;
+      }
+      return true;
+    });
   };
 
+  const buildHeaderLabel = () => {
+    const hasSelection = getSelected(internalValues).length > 0;
+    return labels?.select || !multiSelect ? (
+      <StyledTypography
+        as="div"
+        $selectionDisabled={disabled}
+        variant={isOpen || hasSelection ? "normalText" : "placeholderText"}
+        className={clsx(
+          classes?.placeholder,
+          disabled && classes?.selectionDisabled
+        )}
+      >
+        {selectionLabel.selected}
+      </StyledTypography>
+    ) : (
+      <StyledTypography
+        as="div"
+        $selectionDisabled={disabled}
+        className={clsx(
+          classes?.placeholder,
+          disabled && classes?.selectionDisabled
+        )}
+        variant="normalText"
+      >
+        <b>{selectionLabel.selected}</b>
+        {` ${labels?.multiSelectionConjunction} ${selectionLabel.total}`}
+      </StyledTypography>
+    );
+  };
+
+  const hasLabel = label != null;
+  const hasDescription = description != null;
+
+  // the error message area will only be created if:
+  // - an external element that provides an error message isn't identified via aria-errormessage AND
+  //   - both status and statusMessage properties are being controlled OR
+  //   - status is uncontrolled and required is true
+  const canShowError =
+    ariaErrorMessage == null &&
+    ((status !== undefined && statusMessage !== undefined) ||
+      (status === undefined && required));
+
+  const isStateInvalid = isInvalid(validationState);
+
+  let errorMessageId;
+  if (isStateInvalid) {
+    errorMessageId = canShowError
+      ? setId(elementId, "error")
+      : ariaErrorMessage;
+  }
+
   return (
-    <div ref={ref} className={className} style={{ position: "relative" }}>
-      <DropdownHeader isOpen={isOpen} onClick={onToggleHandler}>
-        <HvTypography
-          variant="label"
-          css={{
-            color: theme.colors.acce4,
+    <StyledHvFormElement
+      id={id}
+      name={name}
+      status={validationState}
+      disabled={disabled}
+      readOnly={readOnly}
+      required={required}
+      className={clsx(dropdownClasses.root, className, classes?.root)}
+      $selectionDisabled={disabled}
+      {...others}
+    >
+      {(hasLabel || hasDescription) && (
+        <StyledLabelContainer className={classes?.labelContainer}>
+          {hasLabel && (
+            <StyledLabel
+              id={setId(elementId, "label")}
+              label={label}
+              className={clsx(classes?.label, classes?.label)}
+            />
+          )}
+
+          {hasDescription && (
+            <HvInfoMessage
+              id={setId(elementId, "description")}
+              className={clsx(classes?.description, classes?.description)}
```

Comment:
The classes are duplicated. I think one of them should be from `dropdownClasses`.